pkg/controller/perconaservermongodb/status.go

```diff
@@ -133,80 +135,42 @@ func (r *ReconcilePerconaServerMongoDB) updateStatus(cr *api.PerconaServerMongoD
 		}
 
 		if mongosStatus.Status != cr.Status.Mongos.Status {
```

Comment:
After you're unconditionally resetting cr.Status.Mongos to nil in [L126](https://github.com/percona/percona-server-mongodb-operator/pull/673/files#diff-4e5365b345441eed738248c78ab6eafc0e04a72f69c46719af42ae8f9b232fdfR126) this check will always pass and mongos would set duplicating conditions.

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/673/commits/19460e39ed16b74058e281d95d75cad81fd158cc